plugins/by-name/yaml-companion/default.nix

```diff
@@ -0,0 +1,53 @@
+{
+  lib,
+  config,
+  ...
+}:
+lib.nixvim.plugins.mkNeovimPlugin {
+  name = "yaml-companion";
+  packPathName = "yaml-companion.nvim";
+  package = "yaml-companion-nvim";
+
+  maintainers = [ lib.maintainers.GaetanLepage ];
+
+  settingsExample = {
+    schemas = [
+      {
+        name = "Argo CD Application";
+        uri = "https://raw.githubusercontent.com/datreeio/CRDs-catalog/main/argoproj.io/application_v1alpha1.json";
+      }
+      {
+        name = "SealedSecret";
+        uri = "https://raw.githubusercontent.com/datreeio/CRDs-catalog/main/bitnami.com/sealedsecret_v1alpha1.json";
+      }
+    ];
+    lspconfig = {
+      settings = {
+        yaml = {
+          format.enable = false;
+          schemaStore = {
+            enable = false;
+            url = "";
+          };
+          schemas = lib.nixvim.nestedLiteralLua "require('schemastore').yaml.schemas()";
+        };
+      };
+    };
+  };
+
+  callSetup = false;
+  extraConfig = cfg: {
+    warnings = lib.nixvim.mkWarnings "plugins.yaml-companion" [
+      {
+        when = !config.plugins.lsp.enable;
+        message = "This plugin requires the `plugins.lsp.servers.yamlls` module to be enabled.";
```

Comment:
Warning condition is checking `plugins.lsp.enable`, while the warning message is referencing `plugins.lsp.servers.yamlls`.

Ideally, we should use `"${opt.foo.bar}"`, as interpolating an actual option will lead to the correctly prefixed option path being used, _and_ it verifies that the option we're referencing actually exists. This is only possible when the option is in in the same level of module eval; i.e. we can't easily reference submodule suboptions.